### DISCUSSION TITLE
[Request] 对于各模型、各平台的支持希望可以以plugin的形式出现。

### DISCUSSION THREAD

**Discussion by C-L-STARK** [lobehub/lobehub#2576] [Development Question]
### 🥰 需求描述

目前项目比较臃肿；可否把对模型、三方平台的支持分离到单独的plugin里。大家使用的时候按需增加到配置里。
比如最轻量级的版本就是仅支持open-ai的基础功能和UI界面。

有需要market或需要其他模型提供商或者其他auth方案的就自己增加plugin依赖到项目里。

### 🧐 解决方案

提供各核心功能以及接口的spi开放给大家就好了。
也可以参考下内部HaloOS BusinessAbility的做法呀、多舒服。

### 📝 补充信息

项目目录里。test文件也混杂在里边，真的很乱糟糟的。
mobile和desktop改起来也费劲，都需要二次确认。为什么不是直接兼容到一个文件里。
可能我也不是特别懂，就是在私有部署的时候浪费了很多时间在修改上边。

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
> 目前项目比较臃肿；可否把对模型、三方平台的支持分离到单独的plugin里。大家使用的时候按需增加到配置里。

不计划

> 提供各核心功能以及接口的spi开放给大家就好了。

不计划

> 也可以参考下内部HaloOS BusinessAbility的做法呀、多舒服。

没听说过

> 项目目录里。test文件也混杂在里边，真的很乱糟糟的。

每个人的习惯不一样而已。你又没有天天对着这个项目开发，站着说话不腰疼 😅

> mobile和desktop改起来也费劲，都需要二次确认。为什么不是直接兼容到一个文件里。

https://github.com/lobehub/lobe-chat/discussions/252 mobile 和 desktop 的拆开的原因在这里，以后建议先看看文档再提问。否则会显得很不专业 🤷‍♂️ 

> 可能我也不是特别懂，就是在私有部署的时候浪费了很多时间在修改上边。

为什么我要花精力让你改起来舒服？你是找我们授权了还是给赞助了？有能力的话就从头开始做呗？白嫖还这么多废话 😕

**Reply by C-L-STARK** [reply]
原来开源就是白嫖啊，感谢你的回复。
